operator/v1alpha1/types_kueue.go

```diff
@@ -0,0 +1,324 @@
+// +openshift:compatibility-gen:level=4
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:path=kueue,scope=Cluster
+// +k8s:openapi-gen=true
+// +genclient
+// +genclient:nonNamespaced
+// +kubebuilder:storageversion
+// +kubebuilder:subresource:status
+// +kubebuilder:validation:XValidation:rule="self.metadata.name == 'cluster'",message="olm is a singleton, .metadata.name must be 'cluster'"
+type Kueue struct {
+	metav1.TypeMeta `json:",inline"`
+	// metadata for kueue
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// spec holds user settable values for configuration
+	// +required
+	Spec KueueOperandSpec `json:"spec"`
+	// status holds observed values from the cluster. They may not be overridden.
+	// +optional
+	Status KueueStatus `json:"status,omitempty"`
+}
+
+type KueueOperandSpec struct {
+	operatorv1.OperatorSpec `json:",inline"`
+	// config that is persisted to a config map
+	// +required
+	Config KueueConfiguration `json:"config"`
+}
+
+// +kubebuilder:validation:Enum="NoQueueName;QueueName"
+type ManageJobsWithoutQueueNameOption string
+
+const (
+	// NoQueueName means that all jobs will be gated by Kueue
+	NoQueueName ManageJobsWithoutQueueNameOption = "NoQueueName"
+	// QueueName means that the jobs require a queue label.
+	QueueName ManageJobsWithoutQueueNameOption = "QueueName"
+)
+
+// +kubebuilder:validation:Enum="Enabled;Disabled"
+type EnabledOrDisabled string
+
+const (
+	Enabled  EnabledOrDisabled = "Enabled"
+	Disabled EnabledOrDisabled = "Disabled"
+)
+
+type KueueConfiguration struct {
+	// waitForPodsReady configures gang admission
+	// +optional
+	WaitForPodsReady WaitForPodsReady `json:"waitForPodsReady,omitempty"`
+	// integrations are the types of integrations Kueue will manager
+	// +required
+	Integrations Integrations `json:"integrations"`
+	// featureGates are advanced features for Kueue
+	// +optional
+	FeatureGates map[string]EnabledOrDisabled `json:"featureGates,omitempty"`
+	// resources provides additional configuration options for handling the resources.
+	// Supports https://github.com/kubernetes-sigs/kueue/blob/release-0.10/keps/2937-resource-transformer/README.md
+	// +optional
+	Resources Resources `json:"resources,omitempty"`
+	// manageJobsWithoutQueueName controls whether or not Kueue reconciles
+	// jobs that don't set the annotation kueue.x-k8s.io/queue-name.
+	// Allowed values are NoQueueName and QueueName
+	// Default will be QueueName
+	// +kubebuilder:default=QueueName
+	// +optional
+	ManageJobsWithoutQueueName *ManageJobsWithoutQueueNameOption `json:"manageJobsWithoutQueueName,omitempty"`
+	// managedJobsNamespaceSelector can be used to omit some namespaces from ManagedJobsWithoutQueueName
+	// Only valid if ManagedJobsWithoutQueueName is NoQueueName
+	// +optional
+	ManagedJobsNamespaceSelector *metav1.LabelSelector `json:"managedJobsNamespaceSelector,omitempty"`
```

Comment:
As these both seem related to configuring how jobs are managed by Kueue, what do you think of nesting these into a new type? Something like:
```yaml
spec:
  config:
    managedJobs:
      namespaceSelector: ...
      mode: AnnotatedWithQueueName | NoAnnotation # "mode" isn't a great name but I couldn't come up with something better within a minute or so
```

Reply:
I see the value in this but we are dropping this from our API.